subprojects/core/src/main/java/org/gradle/api/internal/tasks/SnapshotTaskInputsBuildOperationResult.java

```diff
@@ -196,6 +242,11 @@ public void visitInputFileProperties(final InputFilePropertyVisitor visitor) {
                     state.propertyName = entry.getKey();
                     state.propertyHash = fingerprint.getHash();
                     state.propertyNormalizationStrategyIdentifier = fingerprint.getStrategyIdentifier();
+                    Set<PropertyAttribute> propertyAttributes = new TreeSet<>();
```

Comment:
minor: I suggest extracting a method for determining the attributes for a property.